### PR TITLE
Configure dependency exclusions

### PR DESCRIPTION
**Pull request by dmdashenkov**
### Exclutions

Previously, Spine users had to exclude certain transitive dependencies from their runtime classpath manually. This was required due to runtime collisions in the Protobuf class definitions.

The configuration is non-intuitive and ad-hoc-ish:
```gradle
configurations {
    runtimeClasspath.exclude group 'com.google.protobuf', module: 'protobuf-lite'
    testRuntimeClasspath.exclude group 'com.google.protobuf', module: 'protobuf-lite'
}
```

This PR enables the Bootstrap plugin to perform this configuration automatically for all Java projects.

### Generated Resources

Also in this PR we register the generated resource directory in the source sets so that user doesn't have to do that manually:
```gradle
sourceSets {
    main {
        resources.srcDir "$projectDir/generated/main/resources"
    }
    test {
        resources.srcDir "$projectDir/generated/test/resources"
    }
}
```

This PR resolves 2 items from #6.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

File: plugin/src/main/java/io/spine/tools/gradle/Dependency.java [hunk 1]
```diff
@@ -20,34 +20,38 @@
...
-public enum SpineModule {
+public interface Dependency {
 
-    base,
-    client,
-    server;
-
-    private static final String SPINE_PREFIX = "spine-";
+    /**
+     * Obtains the module group ID.
```
Comment (armiol) [inline plugin/src/main/java/io/spine/tools/gradle/Dependency.java]: It's not module anymore. Please update through the changeset.

File: plugin/src/main/java/io/spine/tools/gradle/ProjectDependencyTarget.java [hunk 2]
```diff
@@ -20,32 +20,56 @@
...
+        Configuration mainConfig = configurations.getByName(RUNTIME_CLASSPATH.getValue());
+        exclude(mainConfig, dependency);
+
+        Configuration testConfig = configurations.getByName(TEST_RUNTIME_CLASSPATH.getValue());
+        exclude(testConfig, dependency);
+    }
+
+    private static void exclude(Configuration configuration, Dependency module) {
+        configuration.exclude(ImmutableMap.of(
+                "group", module.groupId(),
```
Comment (armiol) [inline plugin/src/main/java/io/spine/tools/gradle/ProjectDependencyTarget.java]: I think the literals used here deserve to be documented. Please have a Javadoc for this method.

**Review by armiol — CHANGES_REQUESTED** [review]
@dmdashenkov mostly LGTM. Please see my comments.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Review by armiol — APPROVED** [review]

**Pull request merged**